client/app/bundles/course/users/pages/PersonalTimesShow/index.tsx

```diff
@@ -0,0 +1,229 @@
+  Typography,
+} from '@mui/material';
+import sharedConstants from 'lib/constants/sharedConstants';
+import { CourseUserEntity } from 'types/course/courseUsers';
+import { TimelineAlgorithm } from 'types/course/personalTimes';
+import {
+  fetchPersonalTimes,
+  fetchUsers,
+  loadUser,
+  recomputePersonalTimes,
+  updateUser,
+} from '../../operations';
+import {
+  getAllPersonalTimesEntities,
+  getAllUserMiniEntities,
+  getManageCourseUserPermissions,
+  getManageCourseUsersSharedData,
+  getUserEntity,
+} from '../../selectors';
+import UserManagementTabs from '../../components/navigation/UserManagementTabs';
+import SelectCourseUser from '../../components/misc/SelectCourseUser';
+import PersonalTimesTable from '../../components/tables/PersonalTimesTable';
+
+type Props = WrappedComponentProps;
+
+const translations = defineMessages({
+  manageUsersHeader: {
+    id: 'course.users.manage.header',
+    defaultMessage: 'Manage Users',
+  },
+  recomputeSuccess: {
+    id: 'course.users.manage.personalTimes.recompute.success',
+    defaultMessage: 'Successfully recomputed personal times for {name}',
+  },
+  fetchPersonalTimesFailure: {
+    id: 'course.users.manage.personalTimes.fetch.failue',
+    defaultMessage: 'Failed to fetch personal times',
+  },
+  courseUserHeader: {
+    id: 'course.users.manage.personalTimes.header',
+    defaultMessage: 'Course User',
+  },
+  algorithm: {
+    id: 'course.users.manage.personalTimes.algorithm',
+    defaultMessage: 'Algorithm: {algorithm}',
+  },
+  recomputeLabel: {
+    id: 'course.users.manage.personalTimes.recompute',
+    defaultMessage: 'Recompute all times',
+  },
+  updateSuccess: {
+    id: 'course.users.manage.personalTimes.update.success',
+    defaultMessage: 'Successfully updated {name} to timeline "{timeline}"',
+  },
+  learningRate: {
+    id: 'course.users.manage.personalTimes.learningRate.rate',
+    defaultMessage: 'Learning rate: {rate}',
+  },
+  limits: {
+    id: 'course.users.manage.personalTimes.learningRate.limits',
+    defaultMessage: 'Learning rate effective limits: [{min}, {max}]',
+  },
+});
+
+const PersonalTimesShow: FC<Props> = (props) => {
+  const { intl } = props;
+  const [isLoading, setIsLoading] = useState(true);
+  const [isRecomputing, setIsRecomputing] = useState(false);
+  const { userId } = useParams();
+  const users = useSelector((state: AppState) => getAllUserMiniEntities(state));
+  const currentUser = useSelector((state: AppState) =>
+    getUserEntity(state, +userId!),
+  );
+  const personalTimes = useSelector((state: AppState) =>
+    getAllPersonalTimesEntities(state),
+  );
+  const permissions = useSelector((state: AppState) =>
+    getManageCourseUserPermissions(state),
+  );
+  const sharedData = useSelector((state: AppState) =>
+    getManageCourseUsersSharedData(state),
+  );
+  const [timeline, setTimeline] = useState('fixed');
+  const dispatch = useDispatch<AppDispatch>();
+
+  useEffect(() => {
+    dispatch(fetchUsers(false));
```

Comment:
Ah, the rails page allowed selecting any course user. On one hand, can be argued that we might not need to set personal times for staff since they should be able to access everything. However, I can think of a potential use case if we need to set different deadlines for teaching assistants trying to simulate students, and also this page is the only way to recompute personal times.

Do you think it's necessary to have this personal time feature for staff as well? @ekowidianto